src/controller/CHIPDeviceController.cpp

```diff
@@ -582,15 +582,15 @@ uint16_t DeviceController::FindDeviceIndex(NodeId id)
 
 CHIP_ERROR DeviceController::InitializePairedDeviceList()
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    char * buffer  = nullptr;
+    CHIP_ERROR err   = CHIP_NO_ERROR;
+    uint8_t * buffer = nullptr;
 
     VerifyOrExit(mStorageDelegate != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
 
     if (!mPairedDevicesInitialized)
     {
-        constexpr uint16_t max_size = CHIP_MAX_SERIALIZED_SIZE_U64(kNumMaxPairedDevices);
-        buffer                      = static_cast<char *>(chip::Platform::MemoryCalloc(max_size, 1));
+        constexpr uint16_t max_size = sizeof(uint64_t) * kNumMaxPairedDevices;
+        buffer                      = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(max_size));
```

Comment:
Please use MemoryCalloc to help ensure that in bad logic further down, we at least have deterministic 0's in the spots that may be skipped or otherwise incorrectly used, rather than random garbage.

Reply:
@bzbarsky-apple ?